core/trino-main/src/main/java/io/trino/execution/SetColumnTypeTask.java

```diff
@@ -0,0 +1,122 @@
+import static io.trino.metadata.MetadataUtil.createQualifiedObjectName;
+import static io.trino.spi.StandardErrorCode.COLUMN_NOT_FOUND;
+import static io.trino.spi.StandardErrorCode.COLUMN_TYPE_UNKNOWN;
+import static io.trino.spi.StandardErrorCode.TABLE_NOT_FOUND;
+import static io.trino.spi.StandardErrorCode.TYPE_NOT_FOUND;
+import static io.trino.sql.analyzer.SemanticExceptions.semanticException;
+import static io.trino.sql.analyzer.TypeSignatureTranslator.toTypeSignature;
+import static io.trino.type.UnknownType.UNKNOWN;
+import static java.lang.String.format;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+
+public class SetColumnTypeTask
+        implements DataDefinitionTask<SetColumnType>
+{
+    private final Metadata metadata;
+    private final TypeManager typeManager;
+    private final AccessControl accessControl;
+
+    @Inject
+    public SetColumnTypeTask(Metadata metadata, TypeManager typeManager, AccessControl accessControl)
+    {
+        this.metadata = requireNonNull(metadata, "metadata is null");
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        this.accessControl = requireNonNull(accessControl, "accessControl is null");
+    }
+
+    @Override
+    public String getName()
+    {
+        return "ALTER COLUMN";
+    }
+
+    @Override
+    public ListenableFuture<Void> execute(
+            SetColumnType statement,
+            QueryStateMachine stateMachine,
+            List<Expression> parameters,
+            WarningCollector warningCollector)
+    {
+        Session session = stateMachine.getSession();
+        QualifiedObjectName qualifiedObjectName = createQualifiedObjectName(session, statement, statement.getTableName());
+        RedirectionAwareTableHandle redirectionAwareTableHandle = metadata.getRedirectionAwareTableHandle(session, qualifiedObjectName);
+        if (redirectionAwareTableHandle.getTableHandle().isEmpty()) {
+            String exceptionMessage = format("Table '%s' does not exist", qualifiedObjectName);
+            if (metadata.getMaterializedView(session, qualifiedObjectName).isPresent()) {
+                exceptionMessage += ", but a materialized view with that name exists.";
+            }
+            else if (metadata.getView(session, qualifiedObjectName).isPresent()) {
+                exceptionMessage += ", but a view with that name exists.";
+            }
+            if (!statement.isTableExists()) {
+                throw semanticException(TABLE_NOT_FOUND, statement, exceptionMessage);
+            }
+            return immediateVoidFuture();
+        }
+
+        accessControl.checkCanAlterColumn(session.toSecurityContext(), redirectionAwareTableHandle.getRedirectedTableName().orElse(qualifiedObjectName));
+
+        TableHandle tableHandle = redirectionAwareTableHandle.getTableHandle().get();
+        Map<String, ColumnHandle> columnHandles = metadata.getColumnHandles(session, tableHandle);
```

Comment:
Should you make sure the user can see the column here using `accessControl.filterColumns`?